File: Data Structures & Algorithms/subsets/submission-0.java
class Solution {
    List<List<Integer>> result = new ArrayList<>();
    int size = 0;
    int[] numArray;
    public List<List<Integer>> subsets(int[] nums) {
        numArray = nums;
        size = nums.length;
        backtrack(0, new ArrayList<>());
        return result;
    }

    void backtrack(int start, List<Integer> combo) {
        result.add(new ArrayList<>(combo));
        for (int i = start; i < size;i++) {
            combo.add(numArray[i]);
            backtrack(i + 1, combo);
            combo.remove(combo.size() - 1);
        }
    }
}

File: Data Structures & Algorithms/climbing-stairs/submission-0.kt
class Solution {
    fun climbStairs(n: Int): Int {
        var n1 = 0
        var n2 = 1
        var result = n2
        for (i in 1..n) {
            result = n1+n2
            n1 = n2
            n2 = result
        }
        return result
    }
}
